wrench-prefs/src/main/java/com/izettle/wrench/preferences/WrenchPreferences.kt

```diff
@@ -0,0 +1,85 @@
+package com.izettle.wrench.preferences
+
+import android.content.ContentResolver
+import android.content.Context
+import android.net.Uri
+import com.izettle.wrench.core.Bolt
+import com.izettle.wrench.core.Nut
+import com.izettle.wrench.core.WrenchProviderContract
+
+class WrenchPreferences(context: Context) {
+    private val contentResolver: ContentResolver = context.contentResolver
+
+    private fun insertNut(contentResolver: ContentResolver, nut: Nut) {
+        contentResolver.insert(WrenchProviderContract.nutUri(), nut.toContentValues())
+    }
+
+    private fun getBolt(contentResolver: ContentResolver, key: String): Bolt? {
+        val cursor = contentResolver.query(WrenchProviderContract.boltUri(key), null, null, null, null)
+        cursor.use {
+            if (cursor == null) {
+                return null
+            }
+
+            if (cursor.moveToFirst()) {
+                return Bolt.fromCursor(cursor)
+            }
+        }
+
+        return Bolt()
+    }
+
+    private fun insertBolt(contentResolver: ContentResolver, bolt: Bolt): Uri? {
+        return contentResolver.insert(WrenchProviderContract.boltUri(), bolt.toContentValues())
+    }
+
+    fun <T : Enum<T>> getEnum(key: String, type: Class<T>, defValue: T): T {
+        var bolt = getBolt(contentResolver, key) ?: return defValue
+
+        if (bolt.id == 0L) {
+            bolt = bolt.copy(key = key, type = Enum::class.java.name, value = defValue.toString())
+            val uri = insertBolt(contentResolver, bolt)
+            bolt.id = java.lang.Long.parseLong(uri!!.lastPathSegment)
+
+            for (enumConstant in type.enumConstants) {
+                insertNut(contentResolver, Nut(bolt.id, enumConstant.toString()))
+            }
+        }
+
+        return java.lang.Enum.valueOf<T>(type, bolt.value!!)
```

Comment:
```
val value = bolt.value ?: throw IllegalArgumentException("Value should never be null")
return java.lang.Enum.valueOf<T>(type, value)
```

However I think there is probably an easier more readable version of this that could be done.
Maybe @silverpilen or @Freddroid knows?

Reply:
If you have suggestions on how to translate `java.lang.Enum.valueOf<T>(type, value)` into kotlin aswell I'm very open to that. This was one of the things that did not translate well into kotlin and it feels there should be a better way. 

On top of this is felt like it would make sense to rework the typing-system -> rather relying on java.lang.String.class.getSimpleName() we should have the String type defined in wrench-core.

Reply:
yeah, that would make a lot of sense :).
Kotlin has the extension method valueOf, should be doable, did not spend much time figuring out the correct syntax though.

Reply:
I'll leave this as-is for now then. I'm not sure I think that extra line of code necessarily makes the code clearer - it'll crash all the same...